### PR TITLE
[Typing][PEP585 Upgrade][BUAA][101-110] Use standard collections for type hints for 10 files in python/paddle/

### PR DESCRIPTION
**Pull request by Whsjrczr**
### PR Category

User Experience

### PR Types

Devs

### Description

为如下10个文件添加 PEP 563，并进行 PEP 585 升级：
python/paddle/optimizer/lbfgs.py
python/paddle/optimizer/lr.py
python/paddle/optimizer/momentum.py
python/paddle/optimizer/nadam.py
python/paddle/optimizer/optimizer.py
python/paddle/optimizer/radam.py
python/paddle/optimizer/rmsprop.py
python/paddle/optimizer/rprop.py
python/paddle/optimizer/sgd.py
python/paddle/profiler/profiler.py

### CODE REVIEW

File: python/paddle/optimizer/rmsprop.py [hunk 1]
```diff
@@ -15,7 +15,10 @@
 from __future__ import annotations
 
 import warnings
-from typing import TYPE_CHECKING, Sequence
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from collections.abc import Sequence
```
Comment (SigureMo) [inline python/paddle/optimizer/rmsprop.py]: 31 行不是有了么？放到 31 行那里

File: python/paddle/optimizer/rprop.py [hunk 2]
```diff
@@ -15,7 +15,10 @@
 from __future__ import annotations
 
 import warnings
-from typing import TYPE_CHECKING, Sequence
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from collections.abc import Sequence
```
Comment (SigureMo) [inline python/paddle/optimizer/rprop.py]: 同上

Comment (SigureMo) [reply python/paddle/optimizer/rprop.py]: 所有的都有问题，直接参考 https://github.com/PaddlePaddle/Paddle/pull/66968#discussion_r1701844792 进行修改

Comment (Whsjrczr) [reply python/paddle/optimizer/rprop.py]: 收到，已更新

**Review by SigureMo — APPROVED** [review]
LGTMeow

**Pull request merged**